runelite-client/src/main/java/net/runelite/client/plugins/farmingtracker/data/PatchLocation.java

```diff
@@ -0,0 +1,367 @@
+
+	/**
+	 * Lumbridge
+	 */
+	LUMBRIDGE_HOP("Lumbridge", HOP, NULL_8175, new FarmingTick[]{TICK_10}),
+	LUMBRIDGE_TREE("Lumbridge", TREE, NULL_8391, new FarmingTick[]{TICK_40}),
+
+	/**
+	 * Morytania
+	 */
+	MORYTANIA_ALLOTMENT_NW("Morytania(North West)", ALLOTMENT, NULL_8556, new FarmingTick[]{TICK_10}),
+	MORYTANIA_ALLOTMENT_SE("Morytania(South East)", ALLOTMENT, NULL_8557, new FarmingTick[]{TICK_10}),
+	MORYTANIA_FLOWER("Morytania", FLOWER, NULL_7850, new FarmingTick[]{TICK_5}),
+	MORYTANIA_HERB("Morytania", HERB, NULL_8153, new FarmingTick[]{TICK_20}),
+	MORYTANIA_MUSHROOM("Morytania", SPECIAL, NULL_8337, new FarmingTick[]{TICK_80}),
+
+	/**
+	 * Port Sarim
+	 */
+	PORT_SARIM_SPIRIT_TREE("Port Sarim - Spirit tree", SPECIAL, NULL_8338, new FarmingTick[]{TICK_320}),
+
+	/**
+	 * Rimmington
+	 */
+	RIMMINGTON_BUSH("Rimmington", BUSH, NULL_7578, new FarmingTick[]{TICK_20}),
+
+	/**
+	 * Seers' Village
+	 */
+	SEERS_VILLAGE_HOP("Seers' Village", HOP, NULL_8176, new FarmingTick[]{TICK_10}),
+
+	/**
+	 * Tai Bwo Wannai
+	 */
+	TAI_BWO_WANNI_CALQUAT("Tai Bwo Wannai", SPECIAL, NULL_7807, new FarmingTick[]{TICK_160}),
+
+	/**
+	 * Taverley
+	 */
+	TAVERLEY_TREE("Taverley", TREE, NULL_8388, new FarmingTick[]{TICK_40}),
+
+	/**
+	 * Tree Gnome Village
+	 */
+	TREE_GNOME_VILLAGE_FRUIT_TREE("Tree Gnome Village", FRUIT_TREE, NULL_7963, new FarmingTick[]{TICK_160}),
+
+	/**
+	 * Troll Stronghold
+	 */
+	TROLL_STRONGHOLD_HERB("Troll Stronghold", HERB, NULL_18816, new FarmingTick[]{TICK_20}),
+
+	/**
+	 * Varrock
+	 */
+	VARROCK_TREE("Varrock", TREE, NULL_8390, new FarmingTick[]{TICK_40}),
+
+	/**
+	 * Yanille
+	 */
+	YANILLE_HOP("Yanille", HOP, NULL_8173, new FarmingTick[]{TICK_10});
+
+	private final String location;
+	private final PatchType patchType;
+	private final int patchObjectId;
+	private final FarmingTick[] farmingTick;
+
+	public static Set<PatchLocation> findByPatchType(PatchType patchType)
+	{
+		Set<PatchLocation> patchLocations = new HashSet<>();
+
+		for (PatchLocation patchLocation : PatchLocation.values())
+		{
+			if (patchLocation.getPatchType().equals(patchType))
+			{
+				patchLocations.add(patchLocation);
+			}
+		}
+
+		return patchLocations;
+	}
+
+	public static PatchLocation findByPatchObjectId(int objectId)
+	{
+		for (PatchLocation patchLocation : PatchLocation.values())
+		{
+			if (patchLocation.getPatchObjectId() == objectId)
+			{
+				return patchLocation;
+			}
+		}
+
+		return null;
+	}
+
+	public static Set<PatchLocation> findByFarmingTick(FarmingTick farmingTick)
+	{
+		Set<PatchLocation> patchLocations = new HashSet<>();
+
+		for (PatchLocation patchLocation : PatchLocation.values())
+		{
+			if (Arrays.asList(patchLocation.getFarmingTick()).contains(farmingTick))
+			{
+				patchLocations.add(patchLocation);
+			}
+		}
+
+		return patchLocations;
+	}
+
+	public static PatchLocation findByWorldLocation(Client client, QueryRunner queryRunner, int x, int y)
```

Comment:
Is this unused?

Reply:
It's used on line [145](https://github.com/runelite/runelite/pull/1244/files#diff-f3064b389935d6e6c870e919b7b82024R149) in the FarmingTrackerPlugin.